ecosystem/sep-0031.md

```diff
@@ -249,6 +268,51 @@ In the case where the transaction just cannot be completed, return an error resp
 }
 ```
 
+### Fee
+
+The fee endpoint allows an receiving anchor to report the fee that would be charged for processing the payment of an asset. This endpoint is important to allow a sending anchor to accurately report fees to a sending user even when the fee schedule is complex. If a fee can be fully expressed with the `fee_fixed` and `fee_percent` fields in the `/info` response, then an anchor should not implement this endpoint.
+
+```
+GET DIRECT_PAYMENT_SERVER/fee
+```
+
+Request parameters:
+
+Name | Type | Description
+-----|------|------------
+`type` | string | (optional) Type of payment method, optionally described in the `GET /info` endpoint
+`asset_code` | string | Asset code of the payment
+`amount` | float | Amount of the asset that would be sent to the receiving user
+
+Example request:
+
+```
+GET https://api.example.com/fee?asset_code=USD&amount=110.0&type=cash_pickup
+```
+
+On success the endpoint should return `200 OK` HTTP status code and a JSON object with the following fields:
+
+Name | Type | Description
+-----|------|------------
+`fee` | float | The total fee (in units of the asset sent) that would be charged on top of the specified `amount` of `asset_code` sent to the user.
+
+Example response:
+
+```json
+{
+  "fee": 0.013
+}
+```
+
+Every HTTP status code other than `200 OK` will be considered an error. The body should contain error details.
+For example:
+
+```json
+{
+   "error": "This anchor doesn't support the given currency code: USD"
+}
+```
```

Comment:
❓ I don't see mention of this here so I'm not sure if it is the case, but is this endpoint authenticated or not? There could be times where it being authenticated would allow different fee structures for different users based on their existing volumes, or agreements, or the wallet they're using. This might be out of scope and I'm not asking for it, just thinking out loud.